contracts/OptimisticArbitrator.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "@uma/core/contracts/oracle/interfaces/StoreInterface.sol";
+import "@uma/core/contracts/oracle/interfaces/FinderInterface.sol";
+import "@uma/core/contracts/oracle/interfaces/OptimisticOracleV2Interface.sol";
+
+import "@uma/core/contracts/oracle/implementation/Constants.sol";
+
+/**
+ * @title Optimistic Arbitrator
+ * @notice This contract enables assertions to be made to the Optimistic Oracle and assertions to be ratified by the UMA
+ *  Data Verification Mechanism. An assertion consists of using the requestPrice and proposePrice methods of the OptimisticOracle to
+ *  confirm or deny the response to a question. To ratify an assertion, an initial assertion must be made and disputed, such that the
+ *  question is escalated to the DVM, where UMA token holders vote on it.
+ * @dev This contract is stateless and solely encapsulates the DVM's functionality to leverage the Optimistic Arbitrator pattern.
+ * @dev The Optimistic Oracle's functions are called on behalf of the caller, hence this contract will neither hold nor receive funds
+ *  from these actions. The Optimistic Oracle in this design is used to handle all bond payouts based on the outcome of the DVM vote.
+ */
+contract OptimisticArbitrator {
+    using SafeERC20 for IERC20;
+
+    FinderInterface public immutable finder;
+
+    OptimisticOracleV2Interface public immutable oo;
+
+    IERC20 public immutable currency;
+
+    bytes32 public priceIdentifier = "YES_OR_NO_QUERY";
+
+    /**
+     * @notice Constructor.
+     * @param _finderAddress finder to use to get addresses of DVM contracts.
+     * @param _currency collateral token used to pay rewards and fees.
+     */
+    constructor(address _finderAddress, address _currency) {
+        finder = FinderInterface(_finderAddress);
+        currency = IERC20(_currency);
+        oo = OptimisticOracleV2Interface(finder.getImplementationAddress(OracleInterfaces.OptimisticOracleV2));
+    }
+
+    /**
+     * @notice Makes an assertion to the Optimistic Oracle.
+     * @dev The Optimistic Oracle price proposal is submitted on behalf of the caller, therefore only the caller,
+     *  and not the Optimist Arbitrator, will hold or receive funds.
+     * @param timestamp timestamp of the price being requested.
+     * @param ancillaryData ancillary data representing additional args being passed with the price request.
+     * @param assertedPrice price being proposed.
+     * @param reward reward offered to a successful proposer. Note: this can be 0, which could make sense if the caller
+     * expects to not being disputed.
+     * @param bond custom proposal bond to set for request. If set to 0, defaults to the final fee.
```

Comment:
> If set to 0, defaults to the final fee
I don't believe its true as OO allows setting bond to 0 and this is also being used in `assertAndRatify`.

Reply:
only if requester does not call `setBond`. But `_makeAssertion` always explicitly does it even if bond was set to 0

Reply:
> only if requester does not call setBond

If [requester does not cal](https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/oracle/implementation/OptimisticOracleV2.sol#L147)l `setBond()` isn't the `totalBond` defaulted to `2*finalFee`?

Reply:
yes, but this Optimistic Arbitrator contract always calls `setBond`.

Reply:
Updated to: `If set to 0, the bond collected from the caller equals the finalFee.`